docker/src/main/java/org/jclouds/docker/domain/Network.java

```diff
@@ -17,56 +17,103 @@
+import static org.jclouds.docker.internal.NullSafeCopies.copyWithNullOf;
+
 import java.util.List;
 import java.util.Map;
 
+import com.google.auto.value.AutoValue;
+
 import org.jclouds.javax.annotation.Nullable;
 import org.jclouds.json.SerializedNames;
 
-import com.google.auto.value.AutoValue;
-
 @AutoValue
 public abstract class Network {
 
    @AutoValue
    public abstract static class IPAM {
 
-      IPAM() {} // For AutoValue only!
+      IPAM() { }
 
-      @Nullable
-      public abstract String driver();
+      @Nullable public abstract String driver();
 
       public abstract List<Config> config();
 
       @SerializedNames({"Driver", "Config"})
-      public static IPAM create(String driver, List<Config> config) {
-         return new AutoValue_Network_IPAM(driver, copyOf(config));
+      public static IPAM create(@Nullable String driver, List<Config> config) {
+         return builder()
+               .driver(driver)
+               .config(config)
+               .build();
+      }
+
+      public static Builder builder() {
+         return new AutoValue_Network_IPAM.Builder();
+      }
+
+      @AutoValue.Builder
+      public abstract static class Builder {
+
+         public abstract Builder driver(@Nullable String driver);
+
+         public abstract Builder config(List<Config> config);
+
+         abstract List<Config> config();
+
+         abstract IPAM autoBuild();
+
+         public IPAM build() {
+            return config(copyOf(config()))
+                  .autoBuild();
+         }
       }
 
       @AutoValue
       public abstract static class Config {
 
-         Config() {} // For AutoValue only!
+         Config() { }
 
          public abstract String subnet();
 
-         @Nullable
-         public abstract String ipRange();
+         @Nullable public abstract String ipRange();
 
-         @Nullable
-         public abstract String gateway();
+         @Nullable public abstract String gateway();
 
          @SerializedNames({"Subnet", "IPRange", "Gateway"})
-         public static Config create(String subnet, String ipRange, String gateway) {
-            return new AutoValue_Network_IPAM_Config(subnet, ipRange, gateway);
+         public static Config create(String subnet, @Nullable String ipRange, @Nullable String gateway) {
+            return builder()
+                  .subnet(subnet)
+                  .ipRange(ipRange)
+                  .gateway(gateway)
+                  .build();
+         }
+
+         public static Builder builder() {
+            return new AutoValue_Network_IPAM_Config.Builder();
+         }
+
+         @AutoValue.Builder
+         public abstract static class Builder {
+
+            public abstract Builder subnet(String subnet);
+
+            public abstract Builder ipRange(@Nullable String ipRange);
+
+            public abstract Builder gateway(@Nullable String gateway);
+
+            abstract Config autoBuild();
```

Comment:
Hidden build methods are only needed when we want to enforce immutable collections. If no collections are used in the builders, then just provide a public abstract build() method.